Created blog page

madhuka123 opened the pull request.

madhuka123 commented:
I just created a "blog page" in the navigation bar. It allows students to write and submit blogs.

madhuka123 commented:
![Capture](https://github.com/user-attachments/assets/da2c378d-1638-4c71-95cf-8b7e640ccc18)

madhuka123 commented:
Student Name : Munasinghe P M Y Gunasinghe
Task Description – Create Blog page
What You Did – I just created a "blog page" in the navigation bar. It allows students to write and submit blogs.

vaqwerty2 commented:
There is already a PR raised for the blog page. Please reach out to me on teams to sort this.

madhuka123 commented:
I made a few changes. Now we only have one blog page on our website. Now we can add pictures for the blogs. 
Changes as follows;
![image](https://github.com/user-attachments/assets/2f89f732-b60e-45c7-9b6e-bee7df4bb038)
![image](https://github.com/user-attachments/assets/10c28de0-24a9-4792-84f3-c8421d2933d2)
![image](https://github.com/user-attachments/assets/1bf8e623-42bc-4348-a9ed-dd4b419b77e1)

vaqwerty2 commented:
<img width="1376" alt="Screenshot 2025-04-27 at 17 25 00" src="https://github.com/user-attachments/assets/e440c67a-7ef6-4673-8a2b-dbe61f75c612" />

vaqwerty2 reviewed (changes requested):
<img width="1368" alt="Screenshot 2025-04-27 at 17 26 41" src="https://github.com/user-attachments/assets/623af5fc-571d-4d15-8dee-95615f038da4" />

vaqwerty2 reviewed (changes requested):
Hi! I tested the PR again and found crashes on /blogpage/, /publishedblog/, and /adminblogpage/ — all caused by OperationalError: no such column: home_userblogpage.file. The templates are trying to access a file field, but the models file doesn’t have one. Suggest adding a file = models.FileField(...) and running migrations to fix it.

madhuka123 commented:
I'm surprised this works exactly on my localhost.

vaqwerty2 reviewed (changes requested):
Still the same: 

Maybe there is something wrong in your html pages you are creating might be worthwhile to have a look there too.

madhuka123 commented:
The changes I made to the navigation bar are as follows;
"Blog" has a separate "Blog Page" created, and after clicking on it, it shows Create a Blog and Published a Blog.
![image](https://github.com/user-attachments/assets/fcc8e3e6-4b2a-4c13-9962-e16aa4926948)

Created a new one called "admin settings" and added "admin blog review" to it.
![image](https://github.com/user-attachments/assets/3bc06f34-9dfe-4d15-b14d-1331df8208f4)

I added a feature that allows to report published blogs.
![image](https://github.com/user-attachments/assets/1db79e7d-7a4a-402f-8d7d-b2ffe96ec1fb)
![image](https://github.com/user-attachments/assets/d90d14ac-b676-4b7c-8a96-95b9d24e8cf6)

This way, the admin can view the blogs that have been reported.
![image](https://github.com/user-attachments/assets/de68a196-cc7a-40a2-b0d8-8d772ec2078d)

The report for the blog I created is as follows:
![image](https://github.com/user-attachments/assets/3e1cfeab-41a6-4897-9134-53b2ad4d2ab3)

vaqwerty2 reviewed (changes requested):
Still same errors!
Also too many redirects in the header: One thing leads to another then another.
Makes the header too congested after login. Also, I can't access other pages so can't provide feedback on them.

Another suggestion for the fix: The changes made to the models file might not be consistent with file calls, so maybe you need to add the fields there, also evident in the image. home_userblogpage error and no such table in the models.

Since this is a recurring thing that you are not been able to fix, I would urge to talk to our mentor about this.

<img width="1345" alt="Screenshot 2025-05-08 at 01 27 12" src="https://github.com/user-attachments/assets/a2430d1c-4392-408e-a444-ad46ee33954d" />
<img width="1375" alt="Screenshot 2025-05-08 at 01 27 24" src="https://github.com/user-attachments/assets/a4c1dbf2-6c94-4775-ac57-df8abe0336fe" />
<img width="1352" alt="Screenshot 2025-05-08 at 01 27 40" src="https://github.com/user-attachments/assets/8edc1bdf-9228-4cd7-950d-e0dda1760a72" />

madhuka123 commented:
All good mate. I will do those changes.

madhuka123 commented:
The changes made to the "Blog" navigation bar are as follows,
![image](https://github.com/user-attachments/assets/2481faef-4ff6-4062-8afa-16adf8dbcfe1)

I implemented a new feature that allows administrators to download all reported blog entries as a CSV file.
![image](https://github.com/user-attachments/assets/da5217bd-a912-461c-9239-b94020710b14)

vaqwerty2 commented:
Use Yellow text for Dark mode:
<img width="349" alt="Screenshot 2025-05-16 at 12 09 24" src="https://github.com/user-attachments/assets/58c64fd4-f017-4e79-a220-bbd6f8d385d8" />
<img width="325" alt="Screenshot 2025-05-16 at 12 09 51" src="https://github.com/user-attachments/assets/100fe9e7-fbf6-4774-9417-7ccbe4458f91" />
<img width="261" alt="Screenshot 2025-05-16 at 12 12 11" src="https://github.com/user-attachments/assets/a83391f3-4d63-4318-b901-addcdbbbab90" />


Edit blog not visible in dark mode (Use Yellow Text):
<img width="1028" alt="Screenshot 2025-05-16 at 12 11 09" src="https://github.com/user-attachments/assets/62348184-0b43-4141-9124-6db1dc00b525" />
<img width="973" alt="Screenshot 2025-05-16 at 12 13 00" src="https://github.com/user-attachments/assets/3e9a60a9-c673-4c36-a83e-7338dae719a4" />

Most Places Text is not visible in dark mode, use Yellow colour to fix that.

madhuka123 commented:
I have done changes. Now all text parts are visible. There are multiple merges from others into my branch, and many of them have conflicts. I haven’t made these changes myself, so I’m not sure what the other developments are about. Please review and advise, as I don’t have full context on these updates. It is as follows;
![image](https://github.com/user-attachments/assets/0b006922-894a-4a55-be5a-7a91bedf4e5c)

vaqwerty2 reviewed (changes requested):
Last day today, If not corrected this PR will be closed!
Please update if you can.

madhuka123 commented:
Merge Conflicts in PR have been successfully removed.

vaqwerty2 reviewed (approved):
All good!
